src/components/progress-indicators/Shimmer.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react'
+import { View } from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import Animated, { useAnimatedStyle, useSharedValue, withRepeat, withSequence, withTiming } from 'react-native-reanimated'
+
+import { useHandler } from '../../hooks/useHandler'
+import { useLayout } from '../../hooks/useLayout'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+
+type Props = {
+  isShown?: boolean
+}
+
+export const Shimmer = (props: Props) => {
+  const { isShown = true } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const [containerLayout, handleContainerLayout] = useLayout()
+  const containerWidth = containerLayout.width
+
+  const offset = useSharedValue(0)
+
+  const startAnimation = useHandler(() => {
+    const duration = 1500
+    const startPosition = -containerWidth * 3
+    const endPosition = containerWidth
+    offset.value = startPosition
+    offset.value = withRepeat(withSequence(withTiming(startPosition, { duration }), withTiming(endPosition, { duration })), -1, false)
+  })
+
+  React.useEffect(() => {
+    if (containerWidth > 0) startAnimation()
+  }, [startAnimation, containerWidth])
+
+  const animStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ translateX: offset.value }]
+    }
+  })
+
+  return isShown ? (
+    <View style={styles.container} onLayout={handleContainerLayout}>
+      <Animated.View style={[styles.gradientContainer, animStyle]}>
+        <LinearGradient style={styles.gradient} start={{ x: 0, y: 0 }} end={{ x: 1, y: 1 }} colors={['rgba(0,0,0,0)', theme.shimmerBackgroundHighlight]} />
+        <LinearGradient style={styles.gradient} start={{ x: 1, y: 0 }} end={{ x: 0, y: 1 }} colors={['rgba(0,0,0,0)', theme.shimmerBackgroundHighlight]} />
```

Comment:
why two linearGradients going in criss-cross fashion? Can you show a demo video of what this looks like?